internal/controller/drcluster_controller.go

```diff
@@ -663,6 +679,10 @@ func (u *drclusterInstance) generateDRClusterConfig() (*ramen.DRClusterConfig, e
 		return nil, err
 	}
 
+	if clusterID != u.getK8sClusterID() {
```

Comment:
The validation needs to happen in the DRClusterConfig reconciler, where hub sends it a clID as part of the spec, and on the managed cluster we validate it is the same as the one we desire/think it is.